### PR TITLE
return err if callback is localhost

### PR DESCRIPTION
**Pull request by shotasilagadzetaal**

### CODE REVIEW

File: internal/metamorph/server.go [hunk 1]
```diff
@@ -165,6 +165,13 @@ func ValidateCallbackURL(callbackURL string) error {
 		return nil
 	}
 
+	rejectedCallbackUrlSubstrings := viper.GetStringSlice("metamorph.rejectCallbackContaining")
```
Comment (boecklim) [inline internal/metamorph/server.go:168]: I think this validation should happen in `api` service.

We already have here one validation: https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/pkg/api/handler/default.go#L338

File: pkg/api/handler/default.go [hunk 2]
```diff
@@ -332,12 +334,26 @@ func getTransactionOptions(params api.POSTTransactionParams) (*metamorph.Transac
 	return getTransactionsOptions(api.POSTTransactionsParams(params))
 }
 
+func ValidateCallbackURL(callbackURL string) error {
+	_, err := url.ParseRequestURI(callbackURL)
+	if err != nil {
+		return fmt.Errorf("invalid callback URL [%w]", err)
+	}
+
+	rejectedCallbackUrlSubstrings := viper.GetStringSlice("metamorph.rejectCallbackContaining")
```
Comment (boecklim) [inline pkg/api/handler/default.go:343]: We shouldn't use viper here. Like in the other services this setting should be passed as an option https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/pkg/api/handler/default.go#L42

when a new default handler is created https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/cmd/arc/services/api.go#L109

This should be called `viper.GetStringSlice("metamorph.rejectCallbackContaining")` in `api.go`

File: pkg/api/handler/default_test.go [hunk 3]
```diff
@@ -136,6 +136,43 @@ func TestGETHealth(t *testing.T) {
...
+		{
+			name:        "valid callback URL",
+			callbackURL: "http://api.callback.com",
+		},
+		{
+			name:             "blocked url",
+			callbackURL:      "http://localhost",
+			expectedErrorStr: "callback url not acceptable",
+		},
+	}
```
Comment (boecklim) [inline pkg/api/handler/default_test.go:160]: Add a test case which covers this code

```
	for _, substring := range rejectedCallbackUrlSubstrings {
		if strings.Contains(callbackURL, substring) {
			return fmt.Errorf("callback url not acceptable %s", callbackURL)
		}
	}
```

Comment (shotasilagadzetaal) [reply pkg/api/handler/default_test.go:160]: I think it's covered with test case `expectedErrorStr: "callback url not acceptable",`

File: pkg/api/handler/default.go (hunk 2 above)
Comment (shotasilagadzetaal) [reply pkg/api/handler/default.go:343]: For now no option is passed here `handler.NewDefault(logger, metamorphClient, policy)` How would it work? Would I pass function option which, after running, would set local slice of unacceptable callback url strings(read from viper)?

Comment (boecklim) [reply pkg/api/handler/default.go:343]: Yes. Options are already defined here: https://github.com/bitcoin-sv/arc/blob/277ef146f0210d093daeb33438050bdf082657d8/pkg/api/handler/default.go#L42

Slice of unacceptable callback url strings should then be a member of the `*ArcDefaultHandler` - type and passed in as option read from Viper

**Pull request merged**